tests/extension/oneapi_device_global/device_global_queue_array_copy.cpp

```diff
@@ -0,0 +1,148 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides test for array for overloads for queue::copy for device_global
+//
+*******************************************************************************/
+
+#include "../../common/common.h"
+#include "../../common/type_coverage.h"
+#include "device_global_common.h"
+#include "type_pack.h"
+
+#define TEST_NAME device_global_queue_array_copy
+
+namespace TEST_NAMESPACE {
+using namespace sycl_cts;
+using namespace device_global_common_functions;
+
+#if defined(SYCL_EXT_ONEAPI_PROPERTY_LIST) && \
+    defined(SYCL_EXT_ONEAPI_DEVICE_GLOBAL)
+namespace oneapi = sycl::ext::oneapi;
+
+template <typename T>
+struct queue_array_copy_to_dg_kernel;
+
+template <typename T>
+struct queue_array_change_dg_kernel;
+
+template <typename T>
+oneapi::device_global<T> dev_global_out;
+
+/** @brief The function tests that queue copy overloads correctly copy
+ * single element to device_global
+ *  @tparam T Type of underlying struct
+ */
+template <typename T, size_t N>
+void run_test_copy_to_device_global_array(util::logger& log,
+                                          const std::string& type_name) {
+  T data[N];
+  value_helper<T[N]>::change_val(data, 1);
+  const T* src_data = &data[0];
+  size_t num_element = N / 2;
+
+  auto queue = util::get_cts_object::queue();
+
+  auto event = queue.copy(src_data, dev_global_out<T[N]>, 1, num_element);
+  event.wait();
+
+  bool is_copied_correctly = false;
+  {
+    sycl::buffer<bool, 1> is_cop_corr_buf(&is_copied_correctly,
+                                          sycl::range<1>(1));
+    queue.submit([&](sycl::handler& cgh) {
+      auto is_cop_corr_acc =
+          is_cop_corr_buf.template get_access<sycl::access_mode::write>(cgh);
+      cgh.single_task<queue_array_copy_to_dg_kernel<T>>([=] {
+        is_cop_corr_acc[0] =
+            dev_global_out<T[N]>[num_element] == data[num_element];
+      });
+    });
+    queue.wait_and_throw();
+  }
+  if (!is_copied_correctly) {
+    FAIL(log,
+         get_case_description(
+             "Overload of sycl::queue::copy for device_global",
+             "Didn't copy correct element to device_global array", type_name));
+  }
+}
+
+// Creating instance with default constructor
+template <typename T>
+oneapi::device_global<T> dev_global_in;
+
+/** @brief The function tests that queue copy overloads correctly copy
+ * single element from device_global
+ *  @tparam T Type of underlying struct
+ */
+template <typename T, size_t N>
+void run_test_copy_from_device_global_array(util::logger& log,
+                                            const std::string& type_name) {
+  T new_val[N];
+  value_helper<T[N]>::change_val(new_val, 5);
+  T data[N];
+  std::remove_all_extents_t<T>* dst_data = &data[0];
```

Comment:
At the end you could have just `auto` here...

Reply:
Changed to `auto`.